src/metax_api/tests/api/rest/base/views/datasets/write.py

```diff
@@ -375,6 +375,26 @@ def test_draft_is_permanently_deleted_by_enduser(self):
         response = self.client.delete('/rest/datasets/3')
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND, response.status_code)
 
+    ###
+    # Tests for saving drafts
+    ###
+
+    def test_service_users_can_save_draft_datasets(self):
+        ''' Drafts should be saved without preferred identifier '''
+        # test access as a service-user
+        self._use_http_authorization(method='basic', username='metax')
+
+        for i in ('?draft', '?draft=true'):
+            response = self.client.post('/rest/datasets{}'.format(i), self.cr_test_data, format="json")
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+            self.assertFalse('preferred_identifier' in response.data['research_dataset'], response.data)
+            self.assertTrue(response.data['state'] == 'draft', response.data)
+
+        for i in ('', '?draft=false'):
+            response = self.client.post('/rest/datasets{}'.format(i), self.cr_test_data, format="json")
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+            self.assertTrue('preferred_identifier' in response.data['research_dataset'], response.data)
+            self.assertTrue(response.data['state'] == 'published', response.data)
```

Comment:
The test should check that ``preferred_identifier`` is present, as the tests already do, but additionally verify that the value of ``preferred_identifier`` is the same as the value of its ``identifier`` field, i.e. value is a generic UUID, not URN or DOI.

The top loop is unnecessary in the test, since the handling of syntax like ``?draft`` and ``?draft=true`` is tested already elsewhere. For the bottom loop probably its reasonable to be as it is now.

As a nitpick, ``i`` is a poor name for loop variable here, as ``i`` is commonly used to mean "index" in loops.